README.md

```diff
@@ -4,31 +4,73 @@ Zonemaster Web GUI
 ### Purpose
 This module is the Web Interface part of the Zonemaster project. 
 
-### Prerequisites
+### Installation
+
+Follow the detailed [installation instructions](docs/Installation.md).
+
+##### Prerequisites
 Before you install the Zonemaster Web GUI module, you need the
 Zonemaster Engine test framework installed. Please see the
 [Zonemaster Engine installation instructions](https://github.com/zonemaster/zonemaster-engine/blob/master/docs/Installation.md)
 
 And also the the zonemaster-backend module installed. Please see the [Zonemaster
 Backend installation](https://github.com/zonemaster/zonemaster-backend/blob/master/docs/Installation.md)
 
-### Installation
-
-Follow the detailed [installation instructions](docs/Installation.md).
 
-### Configuration 
+##### Configuration 
 
 Text for configuring the backend are found in the [installation
 instructions](docs/Installation.md).
 
+
 ### Contribution
+
+This project was generate with Angular-CLI 1.6.8 and then updating.
+It use the Angular 2+ framework with all its tools.
+
+The source code of the application is available in ``./src/app`` folder.
+The ``app`` folder is structured as follow : 
+```
++-- components
++-- models
+|   +-- alter.ts            : Alert model (level, message)
+|   +-- index.ts            : module export
++-- pipes
+|   +-- filer.pipe.*                : Filter a list based on a string
+|   +-- filer-by-categories.pipe.*  : Filter the list of result by categories
+|   +-- romanize.pipe.*             : Transform latin number to roman number (1 -> I, 2 -> II)
+|   +-- safe-html.pipe.*            : Sanitize text to HTML
++-- services
+|   +-- alert.service.*     : to display alert messages
+|   +-- app.service.*       : to manage app configuration (api url, etc.)
+|   +-- dns-check.service.* : to communicate with the Zonemaster Backend API 
++-- app.*
+```
+
+The ``components`` folder is composed of subfolders that represent the main components.
+Each folder of component contain three files :
+  - ``*.component.css``       : The component's style 
+  - ``*.component.html``      : The component's html 
+  - ``*.component.*.ts``      : The component's typescript 
```

Comment:
As I can see it you have remove the *.component.spec.ts files. Then I guess the text should be

```
  - ``*.component.ts``      : The component's typescript 
```